extension-manifest-v3/src/utils/api.js

```diff
@@ -19,12 +19,47 @@ const ACCOUNT_URL = `https://accountapi.${DOMAIN}/api/v2.1.0`;
 export const COOKIE_DOMAIN = `.${DOMAIN}`;
 const COOKIE_URL = `https://${DOMAIN}`;
 const COOKIE_DURATION = 60 * 60 * 24 * 90; // 90 days in seconds
-const COOKIE_SHORT_DURATION = 60 * 60 * 24; // 1 day in seconds
+const COOKIE_SHORT_DURATION = 60 * 60; // 1 hour in seconds
+let COOKIE_EXPIRATION_DATE_OFFSET = 0;
 
 export const SIGNON_PAGE_URL = `https://signon.${DOMAIN}/`;
 export const CREATE_ACCOUNT_PAGE_URL = `https://signon.${DOMAIN}/register`;
 export const ACCOUNT_PAGE_URL = `https://account.${DOMAIN}/`;
 
+if (__PLATFORM__ === 'safari') {
+  // Safari has two major inconsistency with the specification:
+  // * for cookies.set() the `expirationDate` is in seconds since 2001-01-01T00:00:00Z (instead of beginning of epoch)
+  // * cookies.get() returns the cookie with `expirationDate` in in milliseconds (instead of seconds)
+  //
+  // Below code tries to detect the offset between the two
+  // and use it to convert the expirationDate to seconds
+  try {
+    const cookie = await chrome.cookies.set({
```

Comment:
wonder what's the cost of this test and if there is a way we cache the result between iOS extension restarts. 
Alternatively we could check Safari version which should be cheaper.

Reply:
Ad 1. I don't think calling one API (cookie.set) is faster than another API, as the `chrome.storage.local` must eventually get data from the disk space.

Ad 2. If we set the version, which would it be? What do you think about the future versions? We have no idea when, or if any, they will fix the issue.